src/libraries/System.DirectoryServices/src/System/DirectoryServices/ActiveDirectory/ActiveDirectorySchemaClass.cs

```diff
@@ -1002,12 +1002,12 @@ public Guid SchemaGuid
                     if (_schemaGuidBinaryForm == null)
                     {
                         // get the property from the server
-                        _schemaGuidBinaryForm = (byte[])GetValueFromCache(PropertyManager.SchemaIDGuid, true);
+                        _schemaGuidBinaryForm = (byte[]?)GetValueFromCache(PropertyManager.SchemaIDGuid, true);
                     }
                 }
 
                 // we cache the byte array and create a new guid each time
-                return new Guid(_schemaGuidBinaryForm);
+                return new Guid(_schemaGuidBinaryForm!);
```

Comment:
I'm not sure but I think:
- when isBound == false object is in invalid state so technically we can just let this go through and you will get ArgumentNull inside of Guid.ctor
- if isBound == true then this will never be null since we pass `mustExist = true` to GetValueFromCache.

I'll edit line above to make sure it's clearer the above won't be null for isBound state.